src/python/pants/testutil/rule_runner.py

```diff
@@ -435,15 +469,17 @@ def add_to_build_file(
             relpath if PurePath(relpath).name.startswith("BUILD") else PurePath(relpath, "BUILD")
         )
         mode = "w" if overwrite else "a"
-        return self.create_file(str(build_path), target, mode=mode)
+        return self._create_file(str(build_path), target, mode=mode)
 
-    def write_files(self, files: Mapping[str | PurePath, str]) -> None:
+    def write_files(self, files: Mapping[str | PurePath, str]) -> tuple[str, ...]:
         """Write the files to the build root.
```

Comment:
Probably worth mentioning the return type - it's a tuple of absolute paths, right?